tests/container-queries.test.tsx

```diff
@@ -0,0 +1,367 @@
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenLastCalledWith(
+      expect.objectContaining({ style: { fontSize: 1 * theme.rem } }),
+      {}
+    );
+  });
+
+  it('supports nested container queries and containers', () => {
+    const StyledComponent = styled.Text<{ width: number }>`
+      width: ${(p) => p.width}px;
+      font-size: 2rem;
+      @container (max-width: 800px) {
+        font-size: 1.5rem;
+      }
+    `;
+    const StyledInnerComponent = styled.Text`
+      width: 50px;
+      color: red;
+      @container (max-width: 800px) {
+        color: blue;
+      }
+    `;
+
+    //testing the innerComponent values would mean accessing the children for styledComponent
+    render(
+      <StyledFunctionContainer width={1000}>
+        <StyledComponent width={1000}>
+          <StyledInnerComponent />
+        </StyledComponent>
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenCalledWith(
+      expect.objectContaining({ style: { fontSize: 2 * theme.rem, width: 1000 } }),
+      {}
+    );
+
+    render(
+      <StyledFunctionContainer width={1000}>
+        <StyledComponent width={100}>
+          <StyledInnerComponent />
+        </StyledComponent>
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenCalledWith(
+      expect.objectContaining({ style: { fontSize: 2 * theme.rem, width: 100 } }),
+      {}
+    );
+
+    render(
+      <StyledFunctionContainer width={100}>
+        <StyledComponent width={1000}>
+          <StyledInnerComponent />
+        </StyledComponent>
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenCalledWith(
+      expect.objectContaining({ style: { fontSize: 1.5 * theme.rem, width: 1000 } }),
+      {}
+    );
+
+    render(
+      <StyledFunctionContainer width={100}>
+        <StyledComponent width={100}>
+          <StyledInnerComponent />
+        </StyledComponent>
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenCalledWith(
+      expect.objectContaining({ style: { fontSize: 1.5 * theme.rem, width: 100 } }),
+      {}
+    );
+  });
+
+  it.skip('does not support named containers', () => {
+    const StyledContainerWithName = styled.View`
+      container-name: sidebar;
+    `;
+    try {
+      render(<StyledContainerWithName />);
+      expect(false);
+    } 
+    catch(e) {
+      if((e as Error).message === "Container-name is not currently supported by styled-native-components") {
+        expect(true);
+      } else {
+        expect(false);
+      }
+    }
+
+  const StyledContainerWithNameAndProps = styled.View<{active: boolean}>`
+      container-name: sidebar;
+      font-size: ${(p) => (p.active ? '2rem' : '1.5rem')};
+    `;
+    expect(() => {
+      render(<StyledContainerWithNameAndProps active/>);
+    }).toThrow('Container-name is not currently supported by styled-native-components');
+  });
+});
```

Comment:
I think we are missing a test for a component both being a container and using a container query.